src/librustc_mir/build/expr/into.rs

```diff
@@ -157,11 +157,8 @@ impl<'a,'tcx> Builder<'a,'tcx> {
                     }
 
                     // execute the body, branching back to the test
```

Comment:
we should add at least a comment here explaining that we are (ab)using `destination` to serve as a dummy location